src/main/scala/com/criteo/hive/SecurityUtil.scala

```diff
@@ -0,0 +1,38 @@
+package com.criteo.hive
+
+import java.net.InetAddress
+import java.util.Locale
+
+/**
+  * Some security utils.
+  */
+object SecurityUtil {
+  val HOSTNAME_PATTERN = "_HOST"
+
+  private def getComponents(principalConfig: String): Array[String] = {
+    if (principalConfig == null) {
+      return null
+    }
```

Comment:
Prefer using options.  Also, under what context principalConfig by null?  An example rewrite of this method would be:

```
private def getComponents(principalConfig: Option[String]): Option[Array[String]] =
  principalConfig.map(_.split("[/@]")
```